kotlin-checks-test-sources/src/main/kotlin/checks/PropertyGetterAndSetterUsageCheckSample.kt

```diff
@@ -0,0 +1,150 @@
+package checks
+
+open class PropertyGetterAndSetterUsageCheckSample {
+
+    private class PrivateClassGetterAndSetter {
+        private var size: Int = 42
+//                  ^^^^> {{Property "size"}}
+        fun getSize(): Int = size // Noncompliant {{Convert this getter to a "get()" on the property "size".}}
+//          ^^^^^^^
+        fun setSize(size: Int) {
+//          ^^^^^^^< {{Setter to convert to a "set(value)"}}
+            this.size = size
+        }
+        fun setSize(other: String) { } // Compliant, argument type is "String" instead of "Int"
+    }
+    private class PrivateClassOnlyGetter {
+        private val name: String = "42"
+//                  ^^^^> {{Property "name"}}
+        fun getName(): String = name // Noncompliant {{Convert this getter to a "get()" on the property "name".}}
+//          ^^^^^^^
+    }
+    private class PrivateClassOnlySetter {
+        private var id: String = ""
+        //          ^^> {{Property "id"}}
+        fun setId(id: String) { // Noncompliant {{Convert this setter to a "set(value)" on the property "id".}}
+//          ^^^^^
+            this.id = id
+        }
+    }
+    private class PrivateClassBooleanGetter {
+        private val active: Boolean = true
+//                  ^^^^^^> {{Property "active"}}
+        fun isActive(): Boolean = active // Noncompliant {{Convert this getter to a "get()" on the property "active".}}
+//          ^^^^^^^^
+    }
+    private open class PrivateClassWithoutIssues(
+        private val constructorPrivateVal: String,
+        private var constructorPrivateVar: String
+    ) {
+        private var value1: String = ""
+        private var value2: String = ""
+        private var value3: String? = null
+        private val value4: Boolean = true
+        private var value5: String = ""
+        private var value6: Boolean? = null
+        private val value7: Boolean
+            get() = true
+        private var _value8: Boolean = true
+        private var value8: Boolean
+            get() = _value8
+            set(value) { _value8 = value }
+        protected var value9: Boolean = true
+        private var away: String = ""
+        private var TER: String = ""
+        fun getConstructorPrivateVal(): String = constructorPrivateVal // Compliant
+        fun getConstructorPrivateVar(): String = constructorPrivateVar // Compliant
+        fun setConstructorPrivateVar(value: String ) { constructorPrivateVar = value } // Compliant
```

Comment:
Why are these examples?

If the implementation of the getter/setter is trivial, you can remove these methods and remove the `private modifier`


If the implementation is not trivial or changes the visibility, okay then you will have to write something like this:

```
private open class PrivateClassWithoutIssues1(constructorPrivateVal: String) {
    var constructorPrivateVal: String = constructorPrivateVal
        private set
    var constructorPrivateVar: String
        get() = constructorPrivateVar
        set(value) { constructorPrivateVar = value }
}
```

And maybe we shouldn't force users to write code like this (that's still questionable). But for default implementation cases, I would like to still report them.

Reply:
As discussed, it will be done later, see [SONARKT-308](https://sonarsource.atlassian.net/browse/SONARKT-308)